probe/controls/controls.go

```diff
@@ -6,33 +6,109 @@ import (
 	"github.com/weaveworks/scope/common/xfer"
 )
 
+// HandlerRegistry is an interface for storing control request
+// handlers.
+type HandlerRegistry interface {
+	// Register a new control handler under a given id.
+	Register(control string, f xfer.ControlHandlerFunc)
+	// Rm deletes the handler for a given name.
+	Rm(control string)
+	// Handler gets the handler for the given id.
+	Handler(control string) (xfer.ControlHandlerFunc, bool)
+	// RegisterBatch adds many controls in one go
+}
+
+type realHandlerRegistry map[string]xfer.ControlHandlerFunc
+
+// Register a new control handler under a given id.
+func (r realHandlerRegistry) Register(control string, f xfer.ControlHandlerFunc) {
+	r[control] = f
+}
+
+// Rm deletes the handler for a given name.
+func (r realHandlerRegistry) Rm(control string) {
+	delete(r, control)
+}
+
+// Handler gets the handler for the given id.
+func (r realHandlerRegistry) Handler(control string) (xfer.ControlHandlerFunc, bool) {
+	handler, ok := r[control]
+	return handler, ok
+}
+
 var (
-	mtx      = sync.Mutex{}
-	handlers = map[string]xfer.ControlHandlerFunc{}
+	mtx                          = sync.Mutex{}
+	realRegistry                 = realHandlerRegistry{}
+	registry     HandlerRegistry = realRegistry
```

Comment:
I'm not sure I understand. This registry was always a singleton, I just kept it like that and added some more functions to the package.

The registry could be a separate object, but I think this could be done in a separate PR.